streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/SavePointServiceImpl.java

```diff
@@ -171,6 +218,182 @@ public SavePoint getLatest(Long id) {
+      }
+    }
+
+    return savepointPath;
+  }
+
+  @Override
+  public void trigger(Long appId, @Nullable String savepointPath) {
+    log.info("Start to trigger savepoint for app {}", appId);
+    Application application = applicationService.getById(appId);
+
+    FlinkRESTAPIWatcher.addSavepoint(application.getId());
+
+    application.setOptionState(OptionState.SAVEPOINTING.getValue());
+    application.setOptionTime(new Date());
+    this.applicationService.updateById(application);
+
+    FlinkEnv flinkEnv = flinkEnvService.getById(application.getVersionId());
+
+    // infer savepoint
+    String customSavepoint = this.getFinalSavepointDir(savepointPath, application);
+
+    FlinkCluster cluster = flinkClusterService.getById(application.getFlinkClusterId());
+    String clusterId = getClusterId(application, cluster);
+
+    Map<String, Object> properties = this.tryGetRestProps(application, cluster);
+
+    TriggerSavepointRequest request =
+        new TriggerSavepointRequest(
+            flinkEnv.getFlinkVersion(),
+            ExecutionMode.of(application.getExecutionMode()),
+            clusterId,
+            application.getJobId(),
+            customSavepoint,
+            application.getK8sNamespace(),
+            properties);
+
+    CompletableFuture<SavepointResponse> savepointFuture =
+        CompletableFuture.supplyAsync(() -> FlinkClient.triggerSavepoint(request), executorService);
+
+    handleSavepointResponseFuture(application, savepointFuture);
+  }
+
+  private void handleSavepointResponseFuture(
+      Application application, CompletableFuture<SavepointResponse> savepointFuture) {
+    CompletableFutureUtils.runTimeout(
+            savepointFuture,
+            10L,
+            TimeUnit.MINUTES,
+            savepointResponse -> {
+              if (savepointResponse != null && savepointResponse.savePointDir() != null) {
+                String savePointDir = savepointResponse.savePointDir();
+                log.info("Request savepoint successful, savepointDir: {}", savePointDir);
+              }
+            },
+            e -> {
+              log.error("Trigger savepoint for flink job failed.", e);
+              ApplicationLog log = new ApplicationLog();
+              log.setAppId(application.getId());
+              log.setYarnAppId(application.getClusterId());
+              log.setOptionTime(new Date());
+              String exception = Utils.stringifyException(e);
+              log.setException(exception);
+              if (!(e instanceof TimeoutException)) {
+                log.setSuccess(false);
+              }
+              applicationLogService.save(log);
+            })
+        .whenComplete(
+            (t, e) -> {
+              application.setOptionState(OptionState.NONE.getValue());
+              application.setOptionTime(new Date());
+              applicationService.update(application);
```

Comment:
after application update need to be call: FlinkRESTAPIWatcher.init()

Reply:
Get it. And I'll reorganize this mechanism about `FlinkRESTAPIWatcher` after the MR.
Thanks for your reminding.